File: api/src/main/java/com/api/app/emum/MEM003.java
package com.api.app.emum;

import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 포인트적립사용사유코드: pointTransactionReasonCode
 * */
@Getter
@AllArgsConstructor
public enum MEM003 implements CommonCode {
    ETC("001", "기타", 1, "365", ""),
    ORDER("002", "주문", 2, "365", ""),
    ;

    private final String code;
    private final String codeName;
    private final int displaySequence;
    private final String referenceValue1;
    private final String referenceValue2;

    public static MEM003 findByCode(String code) {
        return CommonCodeUtil.findByCode(MEM003.class, code);
    }
}

File: api/monolith/src/main/kotlin/com/api/app/emum/MEM003.kt
package com.api.app.emum

/** 포인트거래사유코드: pointTransactionReasonCode */
enum class MEM003(
    override val code: String,
    override val codeName: String,
    override val displaySequence: Int,
    override val referenceValue1: String,
    override val referenceValue2: String
) : CommonCode {
    PURCHASE_EARN("001", "구매적립", 1, "365", ""),
    ORDER("002", "구매사용", 2, "365", ""),
    CANCEL("003", "취소", 3, "365", ""),
    ETC("004", "기타", 4, "365", ""),
    ;

    companion object {
        fun findByCode(code: String): MEM003? = CommonCodeUtil.findByCode(MEM003::class.java, code)
    }
}
